README.md

```diff
@@ -8,21 +8,214 @@
+     --header 'content-type: application/x-www-form-urlencoded' \
+     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
+     ```
+
+   - Windows
+
+     ```bash
+     curl --request POST ^
+     --url https://api.hubapi.com/oauth/v1/token ^
+     --header 'content-type: application/x-www-form-urlencoded' ^
+     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
+     ```
+
+   This command will return the access token necessary for API calls.
+
+   ```json
+   {
+     "token_type": "bearer",
+     "refresh_token": "<Refresh Token>",
+     "access_token": "<Access Token>",
+     "expires_in": 1800
+   }
+   ```
+
+5. Store the access token securely for use in your application.
 
 ## Quickstart
 
-[//]: # (TODO: Add a quickstart guide to demonstrate a basic functionality of the module, including sample code snippets.)
+To use the `HubSpot CRM Leads Connector` in your Ballerina application, update the `.bal` file as follows:
+
+### Step 1: Import the module
+
+Import the `hubspot.crm.objects.leads` module and `oauth2` module.
+
+```ballerina
+import ballerinax/hubspot.crm.objects.leads as hsleads;
+import ballerina/oauth2;
+```
+
+### Step 2: Instantiate a new connector
+
+1. Create a `Config.toml` file and, configure the obtained credentials in the above steps as follows:
+
+   ```toml
+    clientId = "<Client Id>"
+    clientSecret = "<Client Secret>"
+    refreshToken = "<Refresh Token>"
+   ```
+
+2. Instantiate a `hsleads:ConnectionConfig` with the obtained credentials and initialize the connector with it.
+
+    ```ballerina 
+    configurable string clientId = ?;
+    configurable string clientSecret = ?;
+    configurable string refreshToken = ?;
+
+    final hsleads:ConnectionConfig hsleadsConfig = {
+        auth : {
+            clientId,
+            clientSecret,
+            refreshToken,
+            credentialBearer: oauth2:POST_BODY_BEARER
+        }
+    };
+
+    final hsleads:Client hsleads = check new (hsleadsConfig);
+    ```
+
+### Step 3: Invoke the connector operation
+
+Now, utilize the available connector operations. A sample usecase is shown below.
+
+#### Create a Lead
+    
+```ballerina
+public function main() returns error? {
+   SimplePublicObjectInputForCreate payload = {
+      {
+         "associations": [
+            {
+               "types": [
+                  {
+                     "associationCategory": "HUBSPOT_DEFINED",
+                     "associationTypeId":578
+                  }
+               ],
+               "to": {
+                  "id": "YOUR_CONTACT_ID"
+               }
+            }
+         ],
+         "properties": {
+            "hs_lead_name": "John Doe"
+         }
+      }
+   }
+   hsleads:SimplePublicObject createLead = check hsleads->/.post(payload);
+}
+```
```

Comment:
AFAIK there is a 4th step in the `Quickstart` guide for running the Ballerina application. Can you add that section here ?

https://github.com/ballerina-platform/module-ballerinax-twitter/blob/main/ballerina/Module.md#step-4-run-the-ballerina-application